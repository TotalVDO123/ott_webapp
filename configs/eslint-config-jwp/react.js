module.exports = {
  extends: [
    // Extend the base config
    './typescript',

    // Use recommended React rules
    'plugin:react/recommended',
  ],

  rules: {
    // Not needed in React 17
    'react/react-in-jsx-scope': 'off',
  },

  overrides: [
    {
      files: ['*.jsx', '*.tsx', '*.ts'],
      plugins: [
        // Enable linting React code
        'react',
        'react-hooks',
      ],
      rules: {
        // Help with Hooks syntax
        'react-hooks/rules-of-hooks': 'error',
        'react-hooks/exhaustive-deps': 'error',

        // Handled by Typescript
        'react/prop-types': 'off',

        // This rule causes too many false positives, e.g. with default exports or child render function
        'react/display-name': 'off',
      },
    },
  ],

  settings: {
    react: {
      pragma: 'React',
      version: '17',
    },
  },
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
  },
};
