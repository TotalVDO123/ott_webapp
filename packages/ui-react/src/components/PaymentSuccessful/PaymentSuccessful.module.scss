@use '@jwp/ott-ui-react/src/styles/variables';
@use '@jwp/ott-ui-react/src/styles/theme';

.topContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-top: 40px;
  row-gap: 32px;

  .circleHug {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 80px;
    height: 80px;
    border: 2px solid variables.$green;
    border-radius: 50%;
  }
}

.innerContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  row-gap: 8px;

  .title {
    font-weight: var(--body-font-weight-bold);
    font-size: 24px;
  }
  
  .message {
    margin: 0;
    font-size: 16px;
    text-align: center;
  }
}

.buttonContainer {
  align-self: stretch;
}

.welcome {
  h2 {
    font-weight: var(--body-font-weight-bold);
    font-size: 24px;
  }
}
