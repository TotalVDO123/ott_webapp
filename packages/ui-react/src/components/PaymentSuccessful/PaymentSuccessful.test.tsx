import React from 'react';
import { act, fireEvent, render } from '@testing-library/react';

import PaymentSuccessful from './PaymentSuccessful';

describe('<PaymentSuccessful>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<PaymentSuccessful siteName="Sitename!" />);

    expect(container).toMatchSnapshot();
  });

  test('calls the onCloseButtonClick callback when clicking the close button', () => {
    const onCloseButtonClick = vi.fn();
    const { getByText } = render(<PaymentSuccessful onCloseButtonClick={onCloseButtonClick} siteName="Sitename!" />);

    fireEvent.click(getByText('checkout.start_watching'));

    expect(onCloseButtonClick).toBeCalled();
  });

  test('calls the onCloseButtonClick callback when timer expires', () => {
    vi.useFakeTimers();
    const onCountdownCompleted = vi.fn();

    render(<PaymentSuccessful onCountdownCompleted={onCountdownCompleted} siteName="Sitename!" />);

    let i = 10;
    while (i--) {
      act(() => {
        vi.runAllTimers();
      });
    }

    expect(onCountdownCompleted).toBeCalled();
    vi.useRealTimers();
  });
});
