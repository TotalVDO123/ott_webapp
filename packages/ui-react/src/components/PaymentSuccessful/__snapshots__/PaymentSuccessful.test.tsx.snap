// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<PaymentSuccessful> > renders and matches snapshot 1`] = `
<div>
  <div
    class="_topContainer_4874d7"
  >
    <div
      class="_circleHug_4874d7"
    >
      <svg
        fill="none"
        height="65"
        viewBox="0 0 64 65"
        width="64"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13.3333 32.6663L26.6666 45.9997L53.3333 19.333"
          stroke="#5aae4a"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.6"
        />
      </svg>
    </div>
    <div
      class="_innerContainer_4874d7"
    >
      <h2
        class="_title_4874d7"
      >
        checkout.payment_successful_title
      </h2>
      <p
        class="_message_4874d7"
      >
        checkout.payment_successful_description
      </p>
    </div>
    <div
      class="_buttonContainer_4874d7"
    >
      <button
        class="_button_f8f296 _primary_f8f296 _fullWidth_f8f296 _large_f8f296"
        type="button"
      >
        <span>
          checkout.start_watching
        </span>
      </button>
    </div>
  </div>
</div>
`;
